modules/tinymce/src/core/test/ts/browser/ClickContentEditableFalseTest.ts

```diff
@@ -29,20 +29,32 @@ describe('browser.tinymce.core.ClickContentEditableFalseTest', () => {
     TinyAssertions.assertSelection(editor, [], 0, [], 1);
   });
 
+  it('TINY-8169: Click on normal paragraph should not move caret to noneditable', () => {
+    const editor = hook.editor();
+    editor.setContent('<p>a</p><p contenteditable="false">b</p>');
+    TinySelections.setCursor(editor, [ 0, 0 ], 0);
+    clickMiddleOf(editor, [ 0 ]);
+    TinyAssertions.assertCursor(editor, [ 0, 0 ], 0);
+  });
+
+  it('TINY-8169: Click above left of noneditable should place the fake caret before the noneditable', () => {
+    const editor = hook.editor();
+    editor.setContent('<div style="width: 30px; height: 30px; margin-bottom: 10px">a</div><div contenteditable="false">b</div>');
+    clickMiddleOf(editor, [ 0 ], -15, 20);
+    TinyAssertions.assertCursor(editor, [ 1 ], 0);
+    TinyAssertions.assertContentPresence(editor, {
+      'p[data-mce-caret="before"]': 1,
+      'div.mce-visual-caret': 1
+    });
+  });
+
   it('Click on content editable false inside content editable true', () => {
     const editor = hook.editor();
     editor.setContent('<div contenteditable="true"><p contenteditable="false">a</p></div>');
     clickMiddleOf(editor, [ 0, 1 ]);
     TinyAssertions.assertSelection(editor, [ 0 ], 0, [ 0 ], 1);
   });
 
-  it('Click on content editable true inside content editable false', () => {
```

Comment:
Is there a reason this test was deleted?

Reply:
It used to call editor.selection.placeCaretAt to put the caret inside the content editable element this new approach simply lets the browser default execute for this case. So if we want to keep this it would have to be a webdriver click that actually moves the caret. However not sure it's worth the effort.

Reply:
That's fair, I'm just asking as I seem to recall this being a customer reported issue and I was worried about it regressing. Happy to go with whatever you think though 👍

Reply:
I did some digging and the test was added for this fix. https://ephocks.atlassian.net/browse/TINY-1001 and commit here: https://github.com/tinymce/tinymce/commit/7745fca2c091c86a4dfb28488f854d7bb3be3383 however the placeCaretAt call has been there before that see: https://github.com/tinymce/tinymce/commit/90e42ea4f6009d7f3808a89cf0167ac3b53c42d5

If I recall correctly it had to do with Chrome not rendering and this related PR suggest that is the case.
https://github.com/tinymce/tinymce/commit/77a24c2ecb554d9f4dda5f0ccf3c3ed27898a1ac

However I could no longer reproduce any caret issues with Chrome at least not on Linux. So I will add this to the QA notes to look for that issue if it still exists. It's very hard to automatically test for since the caret gets placed correctly in the selection model but the actual visual rendering was broken the caret was invisible if I remember correctly I should have added better comments for that one. It has probably been fixed in the browser since the fix was added 7 years ago.

Reply:
Sounds good, thanks for digging that up Spocke!